src/cloud-api-adaptor/test/e2e/azure_common.go

```diff
@@ -57,5 +57,34 @@ func (c AzureCloudAssert) HasPodVM(t *testing.T, id string) {
 
 func (c AzureCloudAssert) GetInstanceType(t *testing.T, podName string) (string, error) {
 	// Get Instance Type of PodVM
+
+	pod_vm_prefix := "podvm-" + podName
+	pager := pv.AzureProps.ManagedVmClient.NewListPager(pv.AzureProps.ResourceGroupName, nil)
+
+	for pager.More() {
+		page, err := pager.NextPage(context.Background())
+		if err != nil {
+			log.Errorf("failed to advance page: %v", err)
+			return "", err
+		}
+
+		for _, vm := range page.Value {
+			if strings.HasPrefix(*vm.Name, pod_vm_prefix) {
+				// VM found
+				// Extract the VM size
+				if vm.Properties != nil && vm.Properties.HardwareProfile != nil {
+					log.Infof("The VM size for VM '%s' is: %s\n", podName, *vm.Properties.HardwareProfile.VMSize)
+					return string(*vm.Properties.HardwareProfile.VMSize), nil
+				} else {
+					log.Errorf("Failed to get VM size for VM '%s'", podName)
+					return "", nil
+				}
+			}
+		}
+	}
```

Comment:
it might make sense to have a common `findVM` helper function to share code with the `checkVMExistence()` fn.

Reply:
Good suggestion. Updated the commit